aws-frauddetector-detector/src/aws_frauddetector_detector/tests/helpers/test_api_helpers.py

```diff
@@ -4,11 +4,73 @@
 )
 from .. import unit_test_utils
 from unittest.mock import MagicMock
+from botocore.exceptions import ClientError
 
 NUMBER_OF_ITEMS_PER_PAGE = 2
 MAX_PAGES = 10
 
 
+def test_retry_not_found_exceptions_decorator_happy_case():
+    mock_afd_client = unit_test_utils.create_mock_afd_client()
+    mock_afd_client.get_labels = MagicMock()
+    mock_afd_client.exceptions.ResourceNotFoundException = ClientError
+    mock_afd_client.get_labels.side_effect = [
+        ClientError({'Code': '', 'Message': ''}, 'get_labels'),
+        {
+            'labels': [{'property': f'{j}'} for j in range(NUMBER_OF_ITEMS_PER_PAGE)]
+        }
+    ]
+
+    @api_helpers.retry_not_found_exceptions
+    def call_get_labels(frauddetector_client):
+        return frauddetector_client.get_labels()
+
+    response = call_get_labels(mock_afd_client)
+    assert len(response['labels']) == NUMBER_OF_ITEMS_PER_PAGE
+    assert mock_afd_client.get_labels.call_count == 2
+
+
+def test_retry_not_found_exceptions_decorator_no_exception():
+    mock_afd_client = unit_test_utils.create_mock_afd_client()
+    mock_afd_client.get_labels = MagicMock(return_value={
+        'labels': [{'property': f'{j}'} for j in range(NUMBER_OF_ITEMS_PER_PAGE)]
+    })
+
+    @api_helpers.retry_not_found_exceptions
+    def call_get_labels(frauddetector_client):
+        return frauddetector_client.get_labels()
+
+    response = call_get_labels(mock_afd_client)
+    assert len(response['labels']) == NUMBER_OF_ITEMS_PER_PAGE
+    assert mock_afd_client.get_labels.call_count == 1
+
+
+def test_retry_not_found_exceptions_decorator_multi_page():
+    mock_afd_client = unit_test_utils.create_mock_afd_client()
+    mock_afd_client.get_labels = MagicMock()
+    mock_afd_client.exceptions.ResourceNotFoundException = ClientError
+    side_effects = [
+        ClientError({'Code': '', 'Message': ''}, 'get_labels')
+    ]
+    side_effects.extend([
+        {
+            'nextToken': 'notNone',
+            'labels': [{'property': f'{i}.{j}'} for j in range(NUMBER_OF_ITEMS_PER_PAGE)]
+        } for i in range(MAX_PAGES + 10)
+    ])
+    mock_afd_client.get_labels.side_effect = side_effects
+
+    @api_helpers.retry_not_found_exceptions
+    @api_helpers.paginated_api_call('labels', max_pages=MAX_PAGES)
+    @api_helpers.api_call_with_debug_logs
+    def call_get_labels(frauddetector_client, nextToken=None):
+        return frauddetector_client.get_labels(nextToken)
+
+    response = call_get_labels(mock_afd_client)
+    assert len(response['labels']) == NUMBER_OF_ITEMS_PER_PAGE * MAX_PAGES
+    assert mock_afd_client.get_labels.call_count == MAX_PAGES + 1
```

Comment:
Why `MAX_PAGES + 1` here?

Reply:
When calling get_labels here, first there is a RNF error, so we sleep 1 second and try again. The second call is the first call with nextToken = None, and we only call `MAX_PAGES` times, so that results in a maximum of `MAX_PAGES + 1` (since this is also decorated with paginated api call decorator)